pyproject.toml

```diff
@@ -0,0 +1,42 @@
+[build-system]
+requires = ["setuptools", "setuptools-scm"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "lit-gpt"
+version = "0.1.0"
+description = "Open source large language model implementation"
+license = {text = "Apache-2.0"}
+authors = [
+    { name = "Lightning AI", email = "community@lightning.ai" }
+]
+readme = "README.md"
+requires-python = ">=3.8"
+dependencies = [
+    "lightning @ git+https://github.com/Lightning-AI/lightning@master",
+]
+classifiers = [
+    "Topic :: Text Processing"
+]
+
+[project.optional-dependencies]
+all = [
+    "torch>=2.1",  # note this have not been released yet
+    "jsonargparse[signatures]",  # CLI
+]
+dev = [
```

Comment:
it also requires [lm-evaluation-harness](https://github.com/Lightning-AI/lit-gpt/blob/8f04791d6ba9ce28eef4fe3af6b9672e4445ab67/requirements.txt#L12) for evaluation

Reply:
There are no eval tests, so CI doesn't require it